### PR TITLE
Configure che-server only from environment variables

### PR DESCRIPTION
**Pull request by skabashnyuk**
### What does this PR do?
- Removed default properties extraction from war.
- Removed all usage of ```CHE_LOCAL_CONF_DIR```
- Forced unset of ```CHE_LOCAL_CONF_DIR``` to make sure that Eclipse Che server can be configured only with environment variables.  
- Changed location of default properties from ```/codenvy/che.properties``` to ```/che/che.properties```

### What issues does this PR fix or reference?
Unify the way how che-server image is configured https://github.com/eclipse/che/issues/7094

#### Release Notes
che-server image can be configured only with environment variables.

#### Docs PR
https://github.com/eclipse/che-docs/pull/315

### CODE REVIEW

**Comment by skabashnyuk** [comment]
Test docker
- assembly mode
- repo mode
- image mode

OpenShift in progress

**Comment by skabashnyuk** [comment]
OCP - tested

File: dockerfiles/che/entrypoint.sh [hunk 1]
```diff
@@ -99,10 +98,8 @@ set_environment_variables () {
     CHE_HOME=$(echo /"${CHE_HOME}" | sed  's|\\|/|g' | sed 's|:||g')
   fi
 
-  # Che configuration directory - where che.properties lives
-  if [ -z "${CHE_LOCAL_CONF_DIR}" ]; then
-    export CHE_LOCAL_CONF_DIR="${CHE_HOME}/conf/"
-  fi
+  # All Eclipse Che variables has to be set as environment variables.
+  unset CHE_LOCAL_CONF_DIR
```
Comment (benoitf) [inline dockerfiles/che/entrypoint.sh]: question why do we need to unset if it's not used ?

Comment (skabashnyuk) [reply dockerfiles/che/entrypoint.sh]: To make sure nobody trying to work around it and we have perfect portability.
BTW If it's blocker for you I can remove it.

Comment (riuvshin) [reply dockerfiles/che/entrypoint.sh]: Im ok to remove that unset

Comment (skabashnyuk) [reply dockerfiles/che/entrypoint.sh]: ok. No problem

**Review by ghost — APPROVED** [review]

**Comment by benoitf** [comment]
@skabashnyuk I would remove https://github.com/eclipse/che/blob/che6/core/commons/che-core-commons-inject/src/main/java/org/eclipse/che/inject/CheBootstrap.java#L108
and https://github.com/eclipse/che/blob/che6/core/commons/che-core-commons-inject/src/main/java/org/eclipse/che/inject/CheBootstrap.java#L222-L226
(so no need to unset CHE_LOCAL_CONF_DIR, it will be not be used even if defined)

**Comment by benoitf** [comment]
so even agent should not use that pattern if possible
https://github.com/eclipse/che/blob/che6/assembly/assembly-wsagent-server/src/assembly/setenv.sh#L13 and https://github.com/eclipse/che/blob/che6/assembly/assembly-wsagent-server/src/assembly/setenv.sh#L31

**Comment by skabashnyuk** [comment]
I've talked with @riuvshin  and he suggested not to hurry up. Maybe we may have the situation where it might be useful.  For instance for users who don't use cli or che-server image and natively runs tomcat.

**Comment by benoitf** [comment]
who are these users ?

**Comment by benoitf** [comment]
we could add warning at https://github.com/eclipse/che/blob/che6/core/commons/che-core-commons-inject/src/main/java/org/eclipse/che/inject/CheBootstrap.java#L224 that it will be removed before 6.0 GA

**Comment by skabashnyuk** [comment]
For example https://github.com/eclipse/che/issues/6750

**Comment by skabashnyuk** [comment]
I'm not 100% sure that removing at all CHE_LOCAL_CONF at all is a good idea. 
I'm sure that is not needed for che-server image. But it might be some cases when it can be useful. 
@riuvshin @eivantsov  wdyt?

**Comment by benoitf** [comment]
@skabashnyuk on #6750 it's using the docker image

**Comment by skabashnyuk** [comment]
I mean that it might be other contributors who use che in a different way then we are and we should carefully think before removing some ability completely.  For instance, this pr still give the ability to define CHE_LOCAL_CONF directory and mount it in a container if needed, for urgent cases.
If you don't mind I would decouple this issue https://github.com/eclipse/che/issues/7094 from removing completely  ```CHE_LOCAL_CONF```

**Comment by benoitf** [comment]
I do not say that it should not be done carefully but che6 is allowing to do that. so we may deprecated in 5.x and remove it in 6.x

on contributors, I think that env file is allowing to do the extension, also with custom assemblies you can override default value with properties file as well.

And yes I'm fine to decouple issue but as this `local conf folder` way is the "old way" of configuring I would say it's time to get rid of it definitely.

**Review by riuvshin — APPROVED** [review]

**Review by benoitf — APPROVED** [review]

**Comment by benoitf** [comment]
@skabashnyuk should I create an issue for getting rid of conf folder for all instances ?

**Comment by skabashnyuk** [comment]
@benoitf If you think so - then yes, why no. I'm now 40/60(more no, than yes) about that. An issue will be a good place to elaborate about that.

**Comment by benoitf** [comment]
Hello, I've created https://github.com/eclipse/che/issues/7324, it will be easier to discuss in this issue.

**Pull request merged**